### PR TITLE
feat: Renewal reminder emails and improved subscription notes display (v0.4.8)

### PR DESCRIPTION
**Pull request by bscott**
### Renewal Reminder Emails
- Automatic renewal reminders via email
- Configurable reminder window (default: 7 days)
- Daily background scheduler checks for upcoming renewals
- Beautiful HTML email template with subscription details

### Improved Subscription Notes Display
- Notes now shown in hover tooltip instead of separate row
- Eye icon indicator when notes exist
- Auto-sizing tooltip that matches text width
- Cleaner table layout

## 🔧 Technical Changes
- Added `SendRenewalReminder()` method to EmailService
- Added `GetSubscriptionsNeedingReminders()` method to SubscriptionService
- Implemented background scheduler in main.go
- Added comprehensive test suite (13 test cases)
- Updated subscription templates with tooltip UI

## 🧪 Testing
- All tests passing
- Comprehensive coverage of edge cases and boundary conditions

## 📝 Files Changed
- `cmd/server/main.go` - Scheduler implementation
- `internal/service/email.go` - Email reminder method
- `internal/service/subscription.go` - Reminder query method
- `internal/service/renewal_reminder_test.go` - Test suite
- `templates/subscriptions.html` - Tooltip UI
- `templates/subscription-list.html` - Tooltip UI

See RELEASE_NOTES_v0.4.8.md for full details.

### CODE REVIEW

File: internal/service/email.go [hunk 1]
```diff
@@ -223,3 +223,78 @@ func (e *EmailService) SendHighCostAlert(subscription *models.Subscription) erro
...
+	<div class="container">
+		<h2>Subscription Renewal Reminder</h2>
+		<div class="reminder">
+			<strong>🔔 Reminder:</strong> Your subscription <strong>{{.Name}}</strong> will renew in {{.DaysUntilRenewal}} {{if eq .DaysUntilRenewal 1}}day{{else}}days{{end}}.
+		</div>
+		<div class="subscription-details">
+			<h3>Subscription Details</h3>
+			<div class="detail-row"><span class="label">Name:</span> {{.Name}}</div>
+			<div class="detail-row"><span class="label">Cost:</span> ${{printf "%.2f" .Cost}} {{.Schedule}}</div>
+			<div class="detail-row"><span class="label">Monthly Cost:</span> ${{printf "%.2f" .MonthlyCost}}</div>
```
[the remark this answers is not in the document]
Comment (bscott) [reply internal/service/email.go]: @copilot open a new pull request to apply changes based on [this feedback](https://github.com/bscott/subtrackr/pull/58#discussion_r2530302326)

File: internal/service/email.go [hunk 2]
```diff
@@ -223,3 +223,79 @@ func (e *EmailService) SendHighCostAlert(subscription *models.Subscription) erro
...
```
[the remark this answers is not in the document]
Comment (bscott) [reply internal/service/email.go]: @copilot open a new pull request to apply changes based on [this feedback](https://github.com/bscott/subtrackr/pull/58#discussion_r2530154807)

File: .github/workflows/test-build.yml [hunk 3]
```diff
@@ -53,6 +53,18 @@ jobs:
...
+        run: |
+          if [[ "${{ github.ref }}" == refs/tags/* ]]; then
+            GIT_TAG="${{ github.ref_name }}"
+          else
+            GIT_TAG=$(git describe --tags --abbrev=0 2>/dev/null || echo "dev")
+          fi
+          GIT_COMMIT=$(git rev-parse --short HEAD)
+          echo "tag=$GIT_TAG" >> $GITHUB_OUTPUT
+          echo "commit=$GIT_COMMIT" >> $GITHUB_OUTPUT
+
```
[the remark this answers is not in the document]
Comment (bscott) [reply .github/workflows/test-build.yml:67]: @copilot open a new pull request to apply changes based on [this feedback](https://github.com/bscott/subtrackr/pull/58#discussion_r2530182358)

File: .github/workflows/docker-publish.yml [hunk 4]
```diff
@@ -46,6 +46,18 @@ jobs:
...
+        run: |
+          if [[ "${{ github.ref }}" == refs/tags/* ]]; then
+            GIT_TAG="${{ github.ref_name }}"
+          else
+            GIT_TAG=$(git describe --tags --abbrev=0 2>/dev/null || echo "dev")
+          fi
+          GIT_COMMIT=$(git rev-parse --short HEAD)
+          echo "tag=$GIT_TAG" >> $GITHUB_OUTPUT
+          echo "commit=$GIT_COMMIT" >> $GITHUB_OUTPUT
+
```
[the remark this answers is not in the document]
Comment (bscott) [reply .github/workflows/docker-publish.yml:60]: @copilot open a new pull request to apply changes based on [this feedback](https://github.com/bscott/subtrackr/pull/58#discussion_r2530182394)
